rpcs3/Emu/Cell/RawSPUThread.cpp

```diff
@@ -116,6 +161,44 @@ bool RawSPUThread::write_reg(const u32 addr, const u32 value)
 
 	switch (offset)
 	{
+		
+	case MFC_MSSync_offs:
+	{
+		bool q = false;
+		
+		if (MSSync_reg_left)
+		{
+			while (MSSync_reg_left)
+			{
+				if (mfc_queue[0].MSSync_type == 2)
+				{
+					do_dma_transfer(mfc_proxy[0]);
+					mfc_queue.end_pop();
+					if (mfc_queue.size() == 15)
+					{
+						ch_event_stat |= SPU_EVENT_QV;
+					}
+				}
+			}
+			
+			q = true;
+		}
+		
+		if (!q)
+		{
+			for (u32 i = 0 ; i < mfc_queue.size() ; i++)
+			{
+				if (mfc_queue[i].cmd)
+				{
+					MSSync_reg_left++;
+					mfc_queue[i].MSSync_type = 1; //MSSync from spu register.
```

Comment:
Editing the commands from the spu side is weird, also you would need to use .get_push to access them and save reference to the command as mfc_queue[i] can change between your statments